core/store/src/opener.rs

```diff
@@ -1,8 +1,90 @@
-use crate::{version, Mode, StoreConfig};
+use crate::db::rocksdb::snapshot::{Snapshot, SnapshotError};
+use crate::db::rocksdb::RocksDB;
+use crate::version::{DbVersion, DB_VERSION};
+use crate::{Mode, NodeStorage, StoreConfig, Temperature};
 
 const STORE_PATH: &str = "data";
 
-/// Builder for opening a RocksDB database.
+#[derive(Debug, thiserror::Error)]
+pub enum StoreOpenerError<E: std::fmt::Display> {
+    /// I/O or RocksDB-level error while opening or accessing the database.
+    #[error("{0}")]
+    IO(std::io::Error),
```

Comment:
```suggestion
    IO(#[from] std::io::Error),
```

and you won't need to define `From` manually. Or is it the `E: Display` which is problematic? 

`Io(io::Error)` is also a bit of an anti-pattern -- ideally, you'd include the operation which failed,

```rust
OpeningDatabase(io::Error),
FlushingData(io::Error),
```